src/Analyzers/Core/Analyzers/UseInterpolatedString/AbstractUseInterpolatedStringDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.CodeAnalysis.CodeStyle;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.LanguageService;
+
+namespace Microsoft.CodeAnalysis.UseInterpolatedString;
+
+internal abstract class AbstractUseInterpolatedStringDiagnosticAnalyzer<TSyntaxKind, TExpressionSyntax, TStringLiteralExpressionSyntax>
+    : AbstractBuiltInCodeStyleDiagnosticAnalyzer
+    where TSyntaxKind : struct
```

Comment:
Can this be constrained more specificly?
```suggestion
    where TSyntaxKind : struct, Enum
```